packages/@aws-cdk-testing/framework-integ/test/aws-route53-targets/test/integ.appsync-target.ts

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env node
+import * as appsync from 'aws-cdk-lib/aws-appsync';
+import * as acm from 'aws-cdk-lib/aws-certificatemanager';
+import * as route53 from 'aws-cdk-lib/aws-route53';
+import { App, Stack } from 'aws-cdk-lib';
+import { IntegTest } from '@aws-cdk/integ-tests-alpha';
+import { Construct } from 'constructs';
+import * as targets from 'aws-cdk-lib/aws-route53-targets';
+import { join } from 'path';
+
+type TestStackProps = {
+  certificateArn: string;
+  domainName: string;
+  hostedZoneId: string;
+}
+
+class TestStack extends Stack {
+  constructor(scope: Construct, id: string, props: TestStackProps) {
+    super(scope, id);
+
+    const { domainName, certificateArn, hostedZoneId } = props;
+
+    const certificate = acm.Certificate.fromCertificateArn(this, 'cert', certificateArn);
+
+    const api = new appsync.GraphqlApi(this, 'api', {
+      definition: appsync.Definition.fromFile(join(__dirname, '..', '..', 'aws-appsync', 'test', 'appsync.test.graphql')),
+      domainName: {
+        certificate,
+        domainName,
+      },
+      name: 'api',
+    });
+
+    const zone = route53.HostedZone.fromHostedZoneAttributes(this, 'hosted-zone', {
+      zoneName: domainName,
+      hostedZoneId,
+    });
+
+    new route53.ARecord(this, 'Alias', {
+      zone,
+      target: route53.RecordTarget.fromAlias(new targets.AppSync(api)),
+    });
+  }
+}
+
+const certificateArn = process.env.CDK_INTEG_CERTIFICATE_ARN ?? process.env.CERTIFICATE_ARN;
+if (!certificateArn) throw new Error('For this test you must provide your own CertificateArn as an env var "CERTIFICATE_ARN". See framework-integ/README.md for details.');
+const domainName = process.env.CDK_INTEG_DOMAIN_NAME ?? process.env.DOMAIN_NAME;
+if (!domainName) throw new Error('For this test you must provide your own DomainName as an env var "DOMAIN_NAME". See framework-integ/README.md for details.');
+const hostedZoneId = process.env.CDK_INTEG_HOSTED_ZONE_ID ?? process.env.HOSTED_ZONE_ID;
+if (!hostedZoneId) throw new Error('For this test you must provide your own HostedZoneId as an env var "HOSTED_ZONE_ID". See framework-integ/README.md for details.');
+
+const app = new App();
+const testCase = new TestStack(app, 'aws-cdk-appsync-alias-integ', { certificateArn, domainName, hostedZoneId });
+new IntegTest(app, 'appsync-domain-name', {
+  testCases: [testCase],
+});
```

Comment:
```suggestion
  enableLookups: true,
  stackUpdateWorkflow: false,
});
```

When I previously conducted a test using a domain, configuring this setting registered dummy values for the environment variables used by integ-runner in the template.

The current dummy values in the template seem different from the ones used by integ-runner. How were these values set?

https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/integ-runner/lib/runner/runner-base.ts#L429

Reply:
I manually set the current dummy values -- I didn't realise there was an option to have the test runner set dummy values!

However, if I unset the environment variables and try to rerun the test, I get an error saying the env vars must be set. Is there an extra flag or something I'm meant to use?
![image](https://github.com/user-attachments/assets/0162a129-90fd-4bb8-b710-09921bd2ce88)

Reply:
Did you set your certificate arn to `CERTIFICATE_ARN`?

In the following docs, env name is `CERT_ARN`, but in your code env name is　`CERTIFICATE_ARN`.

https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk-testing/framework-integ/README.md

｀export CERTFICATE_ARN=your_certificate_arn` or fix env name in your code is needed, I think.

Reply:
I'm able to get the tests running and passing locally if I use the real certificate etc., but I thought you were saying the committed snapshot should use the integ-test-defined dummy values.

How do I make the snapshot have the "correct" dummy values?

Reply:
If you run the integ test with the following configuration, I believe the dummy values will remain in the template.

```ts
  enableLookups: true,
  stackUpdateWorkflow: false,
});
```

Reply:
@mazyu36 I've just tried this (I accidentally had `enableLookups: false`, instead of `enableLookups: true`) and the output says that the snapshot will change to the dummy values (as expected), but the test then fails due to not having the environment variables set and the actual snapshot file isn't overwritten...

![image](https://github.com/user-attachments/assets/f349653b-9107-4d76-8829-673d61746cf4)

Any ideas?

To confirm, my code for reading the env vars matches what you linked above (i.e. `CERT_ARN` rather than `CERTIFICATE_ARN`):
```ts
const certificateArn = process.env.CDK_INTEG_CERT_ARN ?? process.env.CERT_ARN;
if (!certificateArn) throw new Error('For this test you must provide your own CertificateArn as an env var "CERT_ARN". See framework-integ/README.md for details.');
const domainName = process.env.CDK_INTEG_DOMAIN_NAME ?? process.env.DOMAIN_NAME;
if (!domainName) throw new Error('For this test you must provide your own DomainName as an env var "DOMAIN_NAME". See framework-integ/README.md for details.');
const hostedZoneId = process.env.CDK_INTEG_HOSTED_ZONE_ID ?? process.env.HOSTED_ZONE_ID;
if (!hostedZoneId) throw new Error('For this test you must provide your own HostedZoneId as an env var "HOSTED_ZONE_ID". See framework-integ/README.md for details.');
```

Reply:
Did you set env values before running the integ test?
It seems that `CERT_ARN` is not exported.

```bash
export HOSTED_ZONE_ID=your_hosted_zone_id
export HOSTED_ZONE_NAME=your_hosted_zone_name
export DOMAIN_NAME=your_domain_name
export CERT_ARN=your_certificate_arn
```